File: src/globals.cpp
#include "globals.hpp"

#include "core/print.hpp"

bool globals::verbose = false;
Array<StringView> globals::paths;

auto globals::init() -> void {
	auto path = StringView(getenv("PATH"));
	for(size_t i = 0; i < path.size(); ++i) {
		auto next = path.view(i, path.size()).find(':');
		if(next == -1) {
			paths.append(path.view(i, path.size()));
			break;
		}

		next += i;
		paths.append(path.view(i, next));
		i = next;
	}
}

File: src/globals.zig
const std = @import("std");

const StringViews = std.ArrayList([]const u8);

const MyshInitError = error{
    NoPath,
};

pub fn init(ally: std.mem.Allocator) !void {
    paths = StringViews.init(ally);
    const path = std.os.getenv("PATH") orelse return MyshInitError.NoPath;
    var iterator = std.mem.tokenize(u8, path, ":");
    while (iterator.next()) |piece| {
        try paths.append(piece);
    }
}

pub fn deinit() void {
    paths.deinit();
}

pub var verbose = false;
pub var paths: StringViews = undefined;
